app/models/gem_dependent.rb

```diff
@@ -0,0 +1,50 @@
+class GemDependent
+  attr_reader :gem_names
+
+  def initialize(gem_names)
+    @gem_names = gem_names
+  end
+
+  def memcached_client
+    Rails.cache
+  end
+
+  def fetch_dependencies
+    dependencies = []
+    keys = @gem_names.map { |g| "deps/v1/#{g}" }
```

Comment:
maybe map this to an associative array of name and key to avoid the `gsub` later?